### PR TITLE
[windows] Add Windows AppLocker Data Stream (MSI and Script)

### PR DESCRIPTION
**Pull request by nicpenning**
- Enhancement

## What does this PR do?

This PR adds the Windows AppLocker MSI and Script data stream which allows the ingestion of those events from the Windows Event Log. This also updates the dashboard with a better title and adds 4 new visualizations to explore the data.

Resolves Part of - https://github.com/elastic/integrations/issues/6979

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

image

### CODE REVIEW

**Comment by nicpenning** [comment]
The appropriate ingest pipelines were lost somewhere, I will fix.

File: packages/windows/_dev/deploy/docker/files/config.yml [hunk 1]
```diff
@@ -219,3 +219,47 @@ rules:
...
+                  "_time": "2021-05-25 13:11:45.000 UTC",
+                  "host": "VAGRANT",
+                  "index": "main",
+                  "linecount": "1",
+                  "max_indextime": "1622471606",
+                  "source": "WinEventLog:Security",
+                  "sourcetype": "XmlWinEventLog:Security",
+                  "splunk_server": "69819b6ce1bd"
+              }
+          }
```
Comment (efd6) [inline packages/windows/_dev/deploy/docker/files/config.yml:265]: Final new line.

File: packages/windows/data_stream/applocker_msi_and_script/agent/stream/httpjson.yml.hbs [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+      value: {{token}}
+{{/if}}
+{{/unless}}
+{{/unless}}
+response.decode_as: application/x-ndjson
+{{#if tags.length}}
+tags:
+{{else if preserve_original_event}}
+tags:
+{{/if}}
```
Comment (efd6) [inline packages/windows/data_stream/applocker_msi_and_script/agent/stream/httpjson.yml.hbs:54]: Are these conditions necessary?

File: packages/windows/data_stream/applocker_msi_and_script/elasticsearch/ingest_pipeline/default.yml [hunk 3]
```diff
@@ -0,0 +1,175 @@
+---
+description: Pipeline for Microsoft-Windows-AppLocker/MSI and Script events
+processors:
+  ## ECS and Event fields.
+
+  - set:
+      field: ecs.version
+      value: '8.0.0'
```
Comment (efd6) [inline packages/windows/data_stream/applocker_msi_and_script/elasticsearch/ingest_pipeline/default.yml:8]: Can we use a more up-to-date version number?

**Review by efd6 — COMMENTED** [review]
Can you provide screen shots showing the dashboard changes.

Also, please make the following change to CODEOWNERS
```diff
diff --git a/.github/CODEOWNERS b/.github/CODEOWNERS
index ddcfae883..aacc76f5a 100644
--- a/.github/CODEOWNERS
+++ b/.github/CODEOWNERS
@@ -243,6 +243,7 @@
 /packages/websphere_application_server @elastic/obs-infraobs-integrations
 /packages/windows @elastic/elastic-agent-data-plane @elastic/security-external-integrations
 /packages/windows/data_stream/applocker_exe_and_dll @elastic/security-external-integrations
+/packages/windows/data_stream/applocker_msi_and_script @elastic/security-external-integrations
 /packages/windows/data_stream/forwarded @elastic/security-external-integrations
 /packages/windows/data_stream/perfmon @elastic/elastic-agent-data-plane
 /packages/windows/data_stream/powershell @elastic/security-external-integrations
```

File: packages/windows/_dev/deploy/docker/files/config.yml (hunk 1 above)
Comment (nicpenning) [reply packages/windows/_dev/deploy/docker/files/config.yml:265]: So this should end with a new line?

File: packages/windows/data_stream/applocker_msi_and_script/agent/stream/httpjson.yml.hbs (hunk 2 above)
Comment (nicpenning) [reply packages/windows/data_stream/applocker_msi_and_script/agent/stream/httpjson.yml.hbs:54]: I am unsure. This was a copy from another Windows integration. I am unsure about what this does in the httpsjson.yml.hbs context.

File: packages/windows/data_stream/applocker_msi_and_script/elasticsearch/ingest_pipeline/default.yml (hunk 3 above)
Comment (nicpenning) [reply packages/windows/data_stream/applocker_msi_and_script/elasticsearch/ingest_pipeline/default.yml:8]: I am all for that, yes.

Comment (nicpenning) [reply packages/windows/data_stream/applocker_msi_and_script/elasticsearch/ingest_pipeline/default.yml:8]: Are there checks that will occur if I bump this to 8.9.0?

File: packages/windows/_dev/deploy/docker/files/config.yml (hunk 1 above)
Comment (nicpenning) [reply packages/windows/_dev/deploy/docker/files/config.yml]: I think I fixed this with a commit

File: packages/windows/data_stream/applocker_msi_and_script/elasticsearch/ingest_pipeline/default.yml (hunk 3 above)
Comment (efd6) [reply packages/windows/data_stream/applocker_msi_and_script/elasticsearch/ingest_pipeline/default.yml:8]: It should just update the version in the documents.

Comment (nicpenning) [reply packages/windows/data_stream/applocker_msi_and_script/elasticsearch/ingest_pipeline/default.yml:8]: Okay, I will bump this up

File: packages/windows/data_stream/applocker_msi_and_script/agent/stream/httpjson.yml.hbs (hunk 2 above)
Comment (efd6) [reply packages/windows/data_stream/applocker_msi_and_script/agent/stream/httpjson.yml.hbs:54]: If it's copied across, let's leave it as is.

Comment (nicpenning) [reply packages/windows/data_stream/applocker_msi_and_script/agent/stream/httpjson.yml.hbs:54]: Okay, easy enough

**Comment by nicpenning** [comment]
> Can you provide screen shots showing the dashboard changes.
> 
> Also, please make the following change to CODEOWNERS
> 
> ```diff
> diff --git a/.github/CODEOWNERS b/.github/CODEOWNERS
> index ddcfae883..aacc76f5a 100644
> --- a/.github/CODEOWNERS
> +++ b/.github/CODEOWNERS
> @@ -243,6 +243,7 @@
>  /packages/websphere_application_server @elastic/obs-infraobs-integrations
>  /packages/windows @elastic/elastic-agent-data-plane @elastic/security-external-integrations
>  /packages/windows/data_stream/applocker_exe_and_dll @elastic/security-external-integrations
> +/packages/windows/data_stream/applocker_msi_and_script @elastic/security-external-integrations
>  /packages/windows/data_stream/forwarded @elastic/security-external-integrations
>  /packages/windows/data_stream/perfmon @elastic/elastic-agent-data-plane
>  /packages/windows/data_stream/powershell @elastic/security-external-integrations
> ```

Do you mean update the .png in the package or add those screenshots to this PR for review?

File: packages/windows/data_stream/applocker_msi_and_script/elasticsearch/ingest_pipeline/default.yml (hunk 3 above)
Comment (nicpenning) [reply packages/windows/data_stream/applocker_msi_and_script/elasticsearch/ingest_pipeline/default.yml]: Done

Comment (nicpenning) [reply packages/windows/data_stream/applocker_msi_and_script/elasticsearch/ingest_pipeline/default.yml]: By the way, the docs were not updated when I ran the build. So my guess is that not much changed from 8.0.0 to 8.9.0 in this dataset.

**Comment by nicpenning** [comment]
I added these:
image

And removed the "-" in the title of the dashboard between [Windows Applocker] - Audited Blocked Applications
image

**Review by efd6 — APPROVED** [review]
LGTM

**Pull request merged**